tls/s2n_server_cert.c

```diff
@@ -37,33 +37,18 @@ int s2n_server_cert_recv(struct s2n_connection *conn)
         S2N_ERROR(S2N_ERR_BAD_MESSAGE);
     }
 
-    int certificate_count = 0;
-    while (s2n_stuffer_data_available(&conn->handshake.io)) {
-        uint32_t certificate_size;
+    struct s2n_cert_public_key public_key;
+    struct s2n_blob cert_chain;
+    cert_chain.data = s2n_stuffer_raw_read(&conn->handshake.io, size_of_all_certificates);
+    cert_chain.size = size_of_all_certificates;
 
-        GUARD(s2n_stuffer_read_uint24(&conn->handshake.io, &certificate_size));
+    GUARD(conn->verify_cert_chain_cb(&cert_chain, &public_key, conn->verify_cert_context));
```

Comment:
This seems to leave the validation that the cert-chain has the proper-format (ie., length <data> length <data> etc.,) up to the author of the call-back.   I think it's probably safer to make that particular check independent of the call back, and error out in-case of malformed cert-formats and then call the user-specified call back?    This does mean parsing the cert twice, and perhaps that's a disincentive.